libnopegl/src/ngpu/format.h

```diff
@@ -95,6 +96,7 @@ enum ngpu_format_feature {
     NGPU_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT            = 1 << 2,
     NGPU_FORMAT_FEATURE_COLOR_ATTACHMENT_BLEND_BIT      = 1 << 3,
     NGPU_FORMAT_FEATURE_DEPTH_STENCIL_ATTACHMENT_BIT    = 1 << 4,
+    NGPU_FORMAT_FEATURE_MAX_ENUM = 0x7FFFFFFF
```

Comment:
kinda unrelated but is it really wise to use enums for bitfields? Since they're bitfields, you're never going to compare them together, and you're also going to combined enum together into an integer all the time. Isn't this going to cause issues?

Basically you end up with fields that are flags as an enum, but the value itself is never going to be an enum:

`enum ngpu_format_feature x = NGPU_FORMAT_FEATURE_SAMPLED_IMAGE_BIT | NGPU_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT` is typically wrong because `x` is *not* an enum.

Reply:
The intent here is not use ngpu_format_feature to store the flags (the same applies to ngpu_buffer_usage). If you look at the diff, they are still stored in uint32_t.

The intent here is to make the code consistent with what Vulkan does (additionally vulkan suffixes the enum with "Bits") and declare a custom VK***Flags for each cases (which is a typedef of VkFlags which itself is a typedef of uint32_t).

Making the "bits" enum anonymous sounds reasonable so no one end up using the enum to store the flags.

Reply:
Added a commit to make the relevant enums anonymous.